R/proActiv.R

```diff
@@ -19,71 +20,40 @@
 #' 
 #'
 #' @export
-#' @return A SummarizedExperiment object with assays giving promoter counts and activity 
-#'   with gene expression stored as column data and promoter gene id mapping stored as 
-#'   row data
+#' @return A SummarizedExperiment object with assays giving promoter counts 
+#'   and activity with gene expression stored as metadata. rowData contains
+#'   promoter metadata and absolute promoter activity summarized across
+#'   conditions (if condition is provided)
 #' 
 #' @examples
 #' 
-#' files <- list.files(system.file('extdata/testdata/bam', package = 'proActiv'), 
-#'                     full.names = TRUE)
-#' proActivFromBAM <- proActiv(files = files,
-#'                             promoterAnnotation  = promoterAnnotation.gencode.v34,
-#'                            fileLabels = NULL,
-#'                            condition = c('A549', 'MCF7'),
-#'                            genome = 'hg38',
-#'                            ncores = 1)
+#' files <- list.files(system.file('extdata/vignette/junctions', 
+#'                        package = 'proActiv'), 
+#'                        full.names = TRUE, pattern = 'replicate5')
+#' result <- proActiv(files = files,
+#'                        promoterAnnotation  = promoterAnnotation.gencode.v34,
+#'                        fileLabels = NULL,
+#'                        ncores = 1)
 #'                            
-#' @import SummarizedExperiment
-#' @import S4Vectors
-#' @importFrom dplyr as_tibble '%>%' group_by mutate row_number 
-#' @importFrom rlang .data
```

Comment:
To comply with Bioconductor guidelines:
- Function length less than 50 
- Line width less than 80 

Split `proActiv` into helper functions
- `parseFile` - performs file checks, checks file type and labels, and whether `genome` argument is provided with file is BAM
- `buildSummarizedExperiment` - calls the 'original' set of functions use to calculate junction read counts and estimate promoter activity 
- `summarizeAcrossCondition` - only executed if `condition` argument is supplied. Summarizes result across condition, and calls `categorizePromoters` which categorizes promoters for each cell line